### ISSUE TITLE
BUG: clean-css fails for this css input.

### ISSUE THREAD

**Issue by xuld** [jakubpawlowicz/clean-css#739]
Clean-css  falls into an endless loop for this css input when source-map is enabled.

```
.error {
    color: red /* Missing } here. */


body {
    color: red;
}
```

```
>cleancss --source-map error.css -o error.min.css
```

I've found the result below:

lib/source-maps/track.js:98

```
while (true) {
      var next = part.indexOf(escapePrefix, cursor);

      if (next == -1) {
        context.column += part.substring(cursor).length;
        break;
      }

      context.column += next - cursor;
      cursor += next - cursor;

      var escaped = part.substring(next, part.indexOf('__', next + 1) + 2); // BUG: escaped is an empty string. 
      var encodedValues = escaped.substring(escaped.indexOf('(') + 1, escaped.indexOf(')')).split(',');
      context.line += ~~encodedValues[0];
      context.column = (~~encodedValues[0] === 0 ? context.column : 0) + ~~encodedValues[1];
      cursor += escaped.length; // cursor is never updated due to escaped.length eqauls 0.
    }
```

cursor is never updated due to escaped eqauls "". next is always 1 by the way.

**Comment by jakubpawlowicz** [comment]
I don't see it happening in the latest version - 3.4.10. Could you please check which version of clean-css are you using?

**Comment by xuld** [comment]
My clean-css version is 3.4.10

I've enabled source-map in this case.

**Comment by jakubpawlowicz** [comment]
Fixed in master, due in 4.0.
